drivers/android/node.rs

```diff
@@ -63,37 +68,30 @@ pub(crate) struct NodeDeath {
     // TODO: Add the moment we're using this for two lists, which isn't safe because we want to
     // remove from the list without knowing the list it's in. We need to separate this out.
     death_links: Links<NodeDeath>,
+    #[pin]
     inner: SpinLock<NodeDeathInner>,
 }
 
 impl NodeDeath {
     /// Constructs a new node death notification object.
-    ///
-    /// # Safety
-    ///
-    /// The caller must call `NodeDeath::init` before using the notification object.
-    pub(crate) unsafe fn new(node: Arc<Node>, process: Arc<Process>, cookie: usize) -> Self {
-        Self {
+    #[allow(clippy::new_ret_no_self)]
```

Comment:
Is there a way to avoid having to disable this for every function where we return `impl InInit<Self>`?

Reply:
Not naming it `new`? I am not sure if we should even name them new, maybe `new_in_place` or something might be more expressive. AFAIK we could add `#![allow(clippy::new_ret_no_self)]` to `lib.rs`, but then it would not warn any longer in other places... Maybe we could get in touch with the clippy folks and ask for some workaround (I am not aware of any currently). There is precedent for this: https://github.com/rust-lang/rust-clippy/issues/3313